src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs

```diff
@@ -85,8 +85,21 @@ private async Task<Solution> MakeStaticAsync(Document document, SyntaxNode root,
             root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             node = root.GetAnnotatedNodes(s_annotationForFixedDeclaration).Single();
             var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
-            var oldModifiersAndStatic = syntaxGenerator.GetModifiers(node).WithIsStatic(true);
+            var oldModifiersAndStatic = syntaxGenerator.GetModifiers(node).WithIsStatic(true).WithIsReadOnly(false);
             var newNode = syntaxGenerator.WithModifiers(node, oldModifiersAndStatic);
+
+            // If the new node has accessors, we also need to remove the 'readonly' modifier from them
+            newNode = newNode.ReplaceNodes(
+                syntaxGenerator.GetAccessors(newNode),
+                (originalNode, rewrittenNode) =>
+                {
+                    var oldModifiers = syntaxGenerator.GetModifiers(rewrittenNode);
+                    if (!oldModifiers.IsReadOnly)
+                        return rewrittenNode;
+
+                    return syntaxGenerator.WithModifiers(rewrittenNode, oldModifiers.WithIsReadOnly(false));
```

Comment:
would be nice if this did the above check anyways.  but i dont' mind how this is written.

Reply:
➡️ Simplified this after checking the implementation of `WithModifiers`